### ISSUE TITLE
HTTPRequest ignores cookies since 3.4.x

### ISSUE THREAD

**Issue by carpikes** [godotengine/godot#55837]
### Godot version

3.4.*

### System information

Linux

### Issue description

The `headers` argument of the signal `request_completed` (HTTPRequest) does not contain anymore cookies since 3.4, this prevents a programmer to handle HTTP sessions in godot (basically getting and sending cookies)

### Steps to reproduce

Create empty 2d scene with HTTPRequest node, connect using the UI the request_completed event to _on_HTTPRequest_request_completed.

Attach a script like this to the root node:
```
extends Node2D
func _ready():
	var body = JSON.print({"username": "...", "password": "..."})
	var error = $HTTPRequest.request("https://website-that-sets-cookies", ["Content-Type: application/json"], false, HTTPClient.METHOD_POST, body)
	if error != OK:
		push_error("An error occurred in the HTTP request.")

func _on_HTTPRequest_request_completed(result, response_code, headers, body):
	print("Result: %s, Code: %s, Headers: %s, Body: %s" % [result, response_code, headers, body.get_string_from_utf8()])
```

The output is different if the project is executed using Godot 3.3.x or Godot 3.4.x.

For example:

Godot 3.3.x (works, contains cookies):
```
Result: 0, Code: 200, Headers: [Server: nginx, Date: Sat, 11 Dec 2021 20:55:21 GMT, Content-Type: text/html; charset=utf-8, Content-Length: 60, Connection: keep-alive, Vary: Cookie, Set-Cookie: session=eyJ1c2VyX201igc; HttpOnly; Path=/, Strict-Transport-Security: max-age=63072000], Body: {"status": "ok", "type": "login", "data": "login-completed"}
```

Godot 3.4.x:
```
Result: 0, Code: 200, Headers: [Content-Type: application/json], Body: {"status": "ok", "type": "login", "data": "login-completed"}
```

Basically, all headers are dropped in Godot 3.4.x

### Minimal reproduction project

Here a project smiliar to the one above, which fetches the google home page, you can run it with both version of godot to see the differences.

[http_test.zip](https://github.com/godotengine/godot/files/7697752/http_test.zip)

Godot 3.3.x:
```
Result: 0, Code: 200, Headers: [Date: Sat, 11 Dec 2021 21:01:55 GMT, Expires: -1, Cache-Control: private, max-age=0, Content-Type: text/html; charset=ISO-8859-1, P3P: CP="This is not a P3P policy! See g.co/p3phelp for more info.", Server: gws, X-XSS-Protection: 0, X-Frame-Options: SAMEORIGIN, Set-Cookie: CONSENT=PENDING+332; expires=Mon, 11-Dec-2023 21:01:55 GMT; path=/; domain=.google.it; Secure, Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43", Accept-Ranges: none, Vary: Accept-Encoding, Transfer-Encoding: chunked], Body: <!doctype html><html itemscope="" itemtype="http://schema.org/WebPage" lang="it"><head><meta content="text/html; charset=UTF-8" http-equiv="Content-Type"><meta content="/logos/doodles/2021/seasonal-holidays-2021-6753651837109324-6752733080595605-cst.gif" itemprop="image"><meta content="Festivit
```

Godot 3.4.x:
```
Result: 0, Code: 200, Headers: [Content-Type: application/json], Body: <!doctype html><html itemscope="" itemtype="http://schema.org/WebPage" lang="it"><head><meta content="text/html; charset=UTF-8" http-equiv="Content-Type"><meta content="/logos/doodles/2021/seasonal-holidays-2021-6753651837109324-6752733080595605-cst.gif" itemprop="image"><meta content="Festivit
```

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/54447.

**Comment by carpikes** [comment]
No, I'm not using HTML5 builds. I'm creating a mobile app.

**Comment by Calinou** [comment]
@carpikes Can you reproduce this in any of the [3.4 betas and RCs](https://downloads.tuxfamily.org/godotengine/3.4/) to determine when the regression started?

**Comment by carpikes** [comment]
Until beta6 (included) everything is working well (as in 3.3.x), from rc1 start behaving as the 3.3.4 stable.

**Comment by carpikes** [comment]
Well, I found probably a typo:
https://github.com/godotengine/godot/blob/3.3/scene/main/http_request.cpp#L386
vs
https://github.com/godotengine/godot/blob/3.4/scene/main/http_request.cpp#L386

`headers` (3.3) vs `p_headers` (3.4)

**Comment by Calinou** [comment]
> Well, I found probably a typo: [`3.3`/scene/main/http_request.cpp#L386](https://github.com/godotengine/godot/blob/3.3/scene/main/http_request.cpp?rgh-link-date=2021-12-11T23%3A02%3A33Z#L386) vs [`3.4`/scene/main/http_request.cpp#L386](https://github.com/godotengine/godot/blob/3.4/scene/main/http_request.cpp?rgh-link-date=2021-12-11T23%3A02%3A33Z#L386)
> 
> `headers` (3.3) vs `p_headers` (3.4)

The function parameter was renamed from `headers` to `p_headers`, so this is expected.

**Comment by carpikes** [comment]
Yup sorry, my fault. In the meantime i ran git bisect to find the commit that changed the behavior, and it's this one:
```
6c4bb3fc4a8af7ffd819d86db59008a702d08067 is the first bad commit
commit 6c4bb3fc4a8af7ffd819d86db59008a702d08067
Author: Fabio Alessandrelli <fabio.alessandrelli@gmail.com>
Date:   Wed Oct 13 15:24:31 2021 +0200

    Revert HTTPRequest gzip compression support.
    
    Partial revert of commit c1135cf0063016ce9abacc23d987becaaef5aa9a.

 doc/classes/HTTPRequest.xml | 24 +++---------
 scene/main/http_request.cpp | 96 +--------------------------------------------
 scene/main/http_request.h   |  9 -----
 3 files changed, 8 insertions(+), 121 deletions(-)
```

**Comment by akien-mga** [comment]
> 3.4.*

That's why it's important to be explicit. The regression exists in `3.4-stable`, but it has been fixed since in 3.4.1 RC 1, RC 2 and the future stable.
